nixos/modules/services/monitoring/prometheus/sachet.nix

```
@@ -0,0 +1,92 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.sachet;
+  configFile = pkgs.writeText "sachet.yml" (builtins.toJSON cfg.configuration);
+
+  # sachet does not allow parsing of env variables in its configuration file
+  runScript = pkgs.writeShellScript "run.sh" ''
+    ${pkgs.envsubst}/bin/envsubst -i "${configFile}" > /tmp/sachet.yaml
+    ${pkgs.prometheus-sachet}/bin/sachet -config /tmp/sachet.yaml -listen-address $1
```

Prefer `exec` here for semantic reasons, and double-quoting `$1` because ShellCheck says to ([SC2086](https://www.shellcheck.net/wiki/SC2086)).

```suggestion
    exec ${pkgs.prometheus-sachet}/bin/sachet -config /tmp/sachet.yaml -listen-address "$1"
```